lib/galaxy/tools/parameters/dynamic_options.py

```diff
@@ -388,7 +388,7 @@ def __init__(self, d_option, elem):
         self.ref_name = elem.get("ref", None)
         self.meta_ref = elem.get("meta_ref", None)
         self.metadata_key = elem.get("key", None)
-        assert self.value is not None or ((self.ref_name is not None or self.meta_ref is not None)and self.metadata_key is not None), ValueError("Required 'value' or 'ref' and 'key' attributes missing from filter")
+        assert self.value is not None or ((self.ref_name is not None or self.meta_ref is not None) and self.metadata_key is not None), ValueError("Required 'value' or 'ref' and 'key' attributes missing from filter")
```

Comment:
I think the docstring also needs updating here, right ? Now it reads as if key was required.

Reply:
I just added a space here, but I think it is wrong anyway, it should be:
```suggestion
        assert self.value is not None or self.ref_name is not None or (self.meta_ref is not None and self.metadata_key is not None), ValueError("Required 'value', or 'ref', or 'meta_ref' and 'key' attributes missing from filter")
```

Reply:
Yes, that looks more correct